src/inmanta/env.py

```diff
@@ -248,17 +246,13 @@ def get_packages_in_working_set(cls, inmanta_modules_only: bool = False) -> dict
         :param inmanta_modules_only: Only return inmanta modules from the working set
         """
         return {
-            packaging.utils.canonicalize_name(dist_info.key): packaging.version.Version(dist_info.version)
-            for dist_info in pkg_resources.working_set
-            if not inmanta_modules_only or dist_info.key.startswith(const.MODULE_PKG_NAME_PREFIX)
+            packaging.utils.canonicalize_name(dist_info.name): packaging.version.Version(dist_info.version)
+            for dist_info in distributions()
+            if not inmanta_modules_only or dist_info.name.startswith(const.MODULE_PKG_NAME_PREFIX)
         }
 
     @classmethod
-    def rebuild_working_set(cls) -> None:
-        pkg_resources.working_set = pkg_resources.WorkingSet._build_master()
-
-    @classmethod
-    def get_dependency_tree(cls, dists: abc.Iterable[str]) -> abc.Set[str]:
+    def get_dependency_tree(cls, dists: abc.Iterable[NormalizedName]) -> abc.Set[NormalizedName]:
         """
         Returns the full set of all dependencies (both direct and transitive) for the given distributions. Includes the
```

Comment:
1. I made this all normalized names, to have mypy check it is correct
2. I still find it difficult to estimate if this is efficient or not, but I'm not going to invest too much into it right now